src/index.ts

```diff
@@ -68,77 +68,74 @@ export function NoopFinalAuth() {
 type InitPreContextIn<T extends HasInitPreContext<any, any>> = Parameters<
   T['initPreContext']
 >[0];
-type InitPreContextOut<T extends HasInitPreContext<any, any>> = ReturnType<
+type InitPreContextReturn<T extends HasInitPreContext<any, any>> = ReturnType<
   T['initPreContext']
 >;
 
 type SanitizeParamsContextIn<
   T extends HasSanitizeParams<any, any>
 > = Parameters<T['sanitizeParams']>[0];
-type SanitizeParamsContextOut<
-  T extends HasSanitizeParams<any, any>
-> = ReturnType<T['sanitizeParams']>;
+type SanitizeParamsReturn<T extends HasSanitizeParams<any, any>> = ReturnType<
+  T['sanitizeParams']
+>;
 
 type SanitizeBodyContextIn<T extends HasSanitizeBody<any, any>> = Parameters<
   T['sanitizeBody']
 >[0];
-type SanitizeBodyContextOut<T extends HasSanitizeBody<any, any>> = ReturnType<
+type SanitizeBodyReturn<T extends HasSanitizeBody<any, any>> = ReturnType<
   T['sanitizeBody']
 >;
 
 type PreAuthorizeContextIn<T extends HasPreAuthorize<any, any>> = Parameters<
   T['preAuthorize']
 >[0];
+type PreAuthorizeReturn<T extends HasPreAuthorize<any, any>> = ReturnType<
+  T['preAuthorize']
+>;
 type PreAuthorizeContextOut<T extends HasPreAuthorize<any, any>> = ReturnType<
   T['preAuthorize']
 >;
-type PreAuthorizeContextOutObjectCase<
-  T extends HasPreAuthorize<any, any>
-> = ReturnType<T['preAuthorize']>;
 type PreAuthorizeContextOutFalseCase<
   T extends HasPreAuthorize<any, any>
 > = false & ReturnType<T['preAuthorize']>;
 
```

Comment:
- [ ] you sure the stuff above is right?

Reply:
yeah you right we missing object & in this case
now all good